ivi-layermanagement-api/test/ilm_control_notification_test.cpp

```diff
@@ -773,16 +773,3 @@ TEST_F(NotificationTest, getNotificationWhenScreenshotDone)
     assertCallbackcalled();
     ASSERT_NE(screenshotData.fd.load(), -1);
 }
-
-TEST_F(NotificationTest, getNotificationWhenScreenshotError)
```

Comment:
This notification is necessary e.g. consider the following use case:
1) ilm_control checks for surface id "X" and send the surface screenshot request to server.
2) Before server could read the above request, client destroys surface id "X".
3) Now server can return the error that screenshot has failed.

Two approaches:
1) If client side checks for surface id then and surface does not exist then this callback can be given.
2) Keep checks only on server side.

Server needs to be contacted even to check existence of surface id. So, it looks like its good to let go of check on client side.

Reply:
Thank @HarshaMM  for hint cases.

Currently, the sequence of surface screenshot:
- Send surface get request, if failed or can't get surface context -> return
- Try to create the shm buffer, if can't return
- Send surface screenshot request

For first approach, test still got failure, notification should come after the calling function. 
For second approach: this always sends the surface screenshot request, even get failures in surface_get and create_shm. It may be not uniform with screenshot. In screenshot, only do the screenshot when screen id existence . Do you think we should change both?

Reply:
Current implementation returns error in case surface does not exist, so callback is not necessary then. 
Now, the test case should not expect callback when screenshot API has failed.  Is there any test case which looks for the return of Error from screenshot API. If not, may be this test case can be modified to check return value accordingly and return success.

Reply:
We have https://github.com/COVESA/wayland-ivi-extension/blob/master/ivi-layermanagement-api/test/ilm_control_test.cpp#L605 for test surface screenshot with invalid id, and https://github.com/COVESA/wayland-ivi-extension/blob/master/ivi-layermanagement-api/test/ilm_control_test.cpp#L569 for screenshot with invalid id.

Reply:
Thanks for the test locations.
For the async API also such a test case can be made, which just looks for error from API.
To really test the callback, a test which can simulate the timing conditions is necessary. This looks rather difficult.
Only concern being, not having a test case for return value check.

Reply:
Created new test case for invalid input id of screenshot and surface screenshot. this should return ILM_FAILED and no receive any notifications.